src/libmongoc/tests/TestSuite.c

```diff
@@ -906,26 +882,48 @@ TestSuite_TestMatchesName (const TestSuite *suite,
 }
 
 
+bool
+test_matches (TestSuite *suite, Test *test)
+{
+   int i;
+   bool matches;
+
+   /* If no match patterns were provided, then assume all match. */
+   if (suite->match_patterns.len == 0) {
+      matches = true;
+   } else {
+      matches = false;
+      for (i = 0; i < suite->match_patterns.len; i++) {
+         char *pattern =
+            _mongoc_array_index (&suite->match_patterns, char *, i);
+         if (TestSuite_TestMatchesName (suite, test, pattern)) {
+            matches = true;
+            break;
```

Comment:
nit: Ok, or just return-- variable "matches" might not be needed at all. But it also seems fine as-is.

Reply:
Updated.